distros/generate.sh

```diff
@@ -9,7 +9,13 @@ set -e
 #	or: ./generate.sh debian-newversion
 #		to create a new folder and a Dockerfile within it
 
-cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+# `readlink -f` does not work on macOS.
+# To get `greadlink`, run `brew install coreutils`.
+if [ "$(uname)" == "Darwin" ]; then
```

Comment:
Alternatively, can we modify it to support `brew install gnu-sed` without `--with-default-names`? The default names thing has a lot of side effects on anything else running on your mac. I'm not comfortable recommending it.